project/npda/templates/partials/visit_form_tab.html

```diff
@@ -0,0 +1,96 @@
+{% load npda_tags %}
+
+<input type="radio"
+  name="visit_tab"
+  role="tab"
+  class="basis-full md:basis-1/3 tab font-bold mb-2 text-black border-b-4 hover:border-b-rcpch_pink border-b-rcpch_light_blue_tint3 border-solid border-0 bg-rcpch_light_blue"
+  aria-label="{{label}} {% if categories|categories_have_errors:visit_instance.errors %} ❗ {% endif %}"
+  {% if checked %} checked {% endif %}
+/>
+<div role="tabpanel" class="tab-content order-1 basis-full rounded-none">
+  <div class="flex align-middle items-center bg-gray-100 p-2 m-2">
+    <span class="mr-2 text-nowrap"><i class="fa-solid fa-list"></i> Skip to section: </span>
+    <ul class="flex gap-2 flex-wrap my-2">
+      {% for field_category in categories|split_by_comma %}
+        <li class="inline-block p-2 font-mono bg-{{ field_category|colour_for_category }} hover:shadow-md hover:underline">
+          <a href="#{{ field_category|cut:" " }}_anchor">
+            {% if field_category|category_has_errors:visit_instance.errors %}
```

Comment:
I spent an hour having a go at replacing the various custom template filters with passing categorised data down from the top form render. But in the end decided it wasn't worth it - I was starting to implement a generic audit platform questionnaire renderer which we should look out outside of each project